### ISSUE TITLE
Different behaviour using optional param with engine

### ISSUE THREAD

**Issue by nickcotton** [railsware/js-routes#275]
Hi, noticing a very slight difference in behaviour when upgrading to v2. Have done my best to explain below with a short comparison. For some reason in v2 I'm getting an extra `/` on some paths when I wasn't under the previous version. Entirely possible this is down to user error but thought I would raise anyway.

`routes.rb`

```ruby
Rails.application.routes.draw do
  constraints(MainDomainPatterns) do
    scope "(:store)", store: /au|gb|us/ do
      mount Planner::Engine, at: "/", as: :planner
    end
  end
end
```

Engine `routes.rb`

```ruby
Planner::Engine.routes.draw do
  constraints(MainDomainPatterns) do
    scope "manage" do
      namespace :api, defaults: { format: :json } do
        post 'events/slugs/suggestions'
	  end
    end
  end
end

```

routes.js - 1.4.9

```js
// planner.api_events_slugs_suggestions => /(:store)/manage/api/events/slugs/suggestions(.:format)
  // function(options)
  planner_api_events_slugs_suggestions_path: Utils.route([["format",false]], {}, [2,[2,[2,[7,"/",false],[1,[3,"store",false],false]],[7,"/",false]],[2,[6,"manage",false],[2,[7,"/",false],[2,[6,"api",false],[2,[7,"/",false],[2,[6,"events",false],[2,[7,"/",false],[2,[6,"slugs",false],[2,[7,"/",false],[2,[6,"suggestions",false],[1,[2,[8,".",false],[3,"format",false]],false]]]]]]]]]]])
```

```js
console.log(Routes.planner_api_events_slugs_suggestions_path())
// /manage/api/events/slugs/suggestions
```

routes.js - 2.0.2
```js
/**
  * Generates rails route to
  * /(:store)/manage/api/events/slugs/suggestions(.:format)
  * @param {object | undefined} options
  * @returns {string} route path
  */
 export const plannerApiEventsSlugsSuggestionsPath = __jsr.r({"format":{"d":"json"}}, [2,[2,[2,[7,"/"],[1,[3,"store"]]],[7,"/"]],[2,[6,"manage"],[2,[7,"/"],[2,[6,"api"],[2,[7,"/"],[2,[6,"events"],[2,[7,"/"],[2,[6,"slugs"],[2,[7,"/"],[2,[6,"suggestions"],[1,[2,[8,"."],[3,"format"]]]]]]]]]]]]]);
```

```js
console.log(plannerApiEventsSlugsSuggestionsPath())
// //manage/api/events/slugs/suggestions
```

**Comment by bogdan** [comment]
Version 2.0.3 👍

**Comment by nickcotton** [comment]
@bogdan Thanks for looking at that so quickly. Most appreciated. Have noticed the routes where we have an engine mounted at "/" are now correct but not others which are mounted inside an optional scope.

i.e. the following routes config is now working correctly

```ruby
Rails.application.routes.draw do
  constraints(WeddingDomainPatterns) do
    mount Planner::Engine, at: "/", as: :wedding
  end
end
```

but this section is still not and did not create any diff between v2.0.2 and v2.0.3 after running `rake js:routes`.

```ruby
Rails.application.routes.draw do
  constraints(MainDomainPatterns) do
    scope "(:store)", store: /au|gb|us/ do
      mount Planner::Engine, at: "/", as: :planner
    end
  end
end
```

Could that be a separate issue? Happy to create a new issue and can provide additional information if you'd like.

**Comment by bogdan** [comment]
@nickcotton fixed it in 8486232e9ac60ee146694914607428fc1c3c4c19. can you check if it works for you?

**Comment by nickcotton** [comment]
@bogdan Yes, have just tested and seems to be working perfectly for all our use cases now. Thanks, this has been really helpful.
